### DISCUSSION TITLE
11ty and S3 - Best Practices For A Workflow/Environment

### DISCUSSION THREAD

**Discussion by chrisativonne** [11ty/eleventy#2011] [Q&A]
I've been building static sites in S3, and just now converting them to 11ty on my local mac, then uploading the files via browser into the S3 bucket. I'm admittedly behind on incorporating something like Git to push/sync my files up.

The background is that I'm migrating from Wordpress to static sites, and I'm trying to find a way that lets me easily edit and test my files in my local browser before committing my changes to the S3 bucket.

What are some best practices and how is your environment/workflow setup? 

Does this possible workflow make sense?:
- Keep my working-in-progress 11ty folder on my local file system
- When I want to add or edit content perform an AWS Cli to sync/download all of my files from the S3 bucket.
- Make my changes. Run eleventy and serve it to the browser to confirm the edits, then
- Use AWS cli commands to push it back up to S3? (Do I even need Github)?

Is there a better way?

**Comment by pdehaan** [comment]
I haven't used S3 much, but I just preview my 11ty sites locally via <kbd>eleventy --serve</kbd> to preview it in a browser/webserver, and then push the generated site filed to GitHub pages or Vercel or Netlify.
I believe each of these servers also work w/ hooks so if you do use git/GitHub, you can just push to a branch and the site should deploy automatically if you configure it.

**Reply by cfjedimaster** [reply]
I'll plus one the above. I've been using Git for years and I'm still a newbie with it. You do not need excessive skills at all to get comfortable enough with it to store your Eleventy site in a repo, and then sign up for Netlify or Vercel and do a deployment whenever you commit your changes.
